simulations/vip-398/arbitrumone.ts

```diff
@@ -0,0 +1,294 @@
+  let converterNetwork: Contract;
+  let xvsVaultTreasury: Contract;
+  let protocolShareReserve: Contract;
+  let usdtPrimeConverter: Contract;
+  let usdcPrimeConverter: Contract;
+
+  before(async () => {
+    usdcPrimeConverter = new ethers.Contract(USDC_PRIME_CONVERTER, SINGLE_TOKEN_CONVERTER_ABI, provider);
+    usdtPrimeConverter = new ethers.Contract(USDT_PRIME_CONVERTER, SINGLE_TOKEN_CONVERTER_ABI, provider);
+
+    converterNetwork = new ethers.Contract(CONVERTER_NETWORK, CONVERTER_NETWORK_ABI, provider);
+    protocolShareReserve = new ethers.Contract(
+      ARBITRUM_PROTOCOL_SHARE_RESERVE_PROXY,
+      PROTOCOL_SHARE_RESERVE_ABI,
+      provider,
+    );
+    xvsVaultTreasury = new ethers.Contract(XVS_VAULT_TREASURY, XVS_VAULT_TREASURY_ABI, provider);
+
+    await pretendExecutingVip(await vip018());
+  });
+
+  describe("Pre-VIP behaviour", () => {
+    before(async () => {
+      prime = new ethers.Contract(ARBITRUM_PRIME, PRIME_ABI, provider);
+    });
+
+    it("Prime should have not contain number of revocable and irrevocable tokens", async () => {
+      expect(await prime.irrevocableLimit()).to.equal(0);
+      expect(await prime.revocableLimit()).to.equal(0);
+    });
+  });
+
+  testForkedNetworkVipCommands("VIP 398", await vip398(), {
+    callbackAfterExecution: async txResponse => {
+      await expectEvents(txResponse, [PRIME_ABI], ["MarketAdded"], [4]);
+      await expectEvents(txResponse, [PRIME_ABI], ["MintLimitsUpdated"], [1]);
+      await expectEvents(txResponse, [PROTOCOL_SHARE_RESERVE_ABI], ["DistributionConfigAdded"], [6]);
+      await expectEvents(txResponse, [PROTOCOL_SHARE_RESERVE_ABI], ["DistributionConfigUpdated"], [2]);
+    },
+  });
+
+  describe("Post-VIP behavior", () => {
+    let prime: Contract;
+    let plp: Contract;
+
+    before(async () => {
+      prime = new ethers.Contract(ARBITRUM_PRIME, PRIME_ABI, provider);
+      plp = new ethers.Contract(ARBITRUM_PLP, PLP_ABI, provider);
+    });
+
+    it("prime should have correct pool registry address", async () => {
+      expect(await prime.poolRegistry()).to.be.equal(arbitrumone.POOL_REGISTRY);
+    });
+
+    it("Comptroller lst and core should have correct Prime token address", async () => {
+      const comptrollerCore = new ethers.Contract(ARBITRUM_COMPTROLLER_CORE, COMPTROLLER_ABI, provider);
+      expect(await comptrollerCore.prime()).to.be.equal(ARBITRUM_PRIME);
+
+      const comptrollerLst = new ethers.Contract(ARBITRUM_COMPTROLLER_LST, COMPTROLLER_ABI, provider);
+      expect(await comptrollerLst.prime()).to.be.equal(ARBITRUM_PRIME);
+    });
+
+    it("Prime should contain correct markets", async () => {
+      expect((await prime.markets(ARBITRUM_VUSDC_CORE))[4]).to.be.equal(true);
+      expect((await prime.markets(ARBITRUM_VUSDT_CORE))[4]).to.be.equal(true);
+      expect((await prime.markets(ARBITRUM_VWBTC_CORE))[4]).to.be.equal(true);
+      expect((await prime.markets(ARBITRUM_VWETH_LST))[4]).to.be.equal(true);
+    });
+
```

Comment:
We could add a check claiming a Prime token, impersonating one of the users who will be able to claim a Prime token as soon as the VIP is executed

Reply:
done, https://github.com/VenusProtocol/vips/pull/429/commits/7c7baa1defabd4b994592d906d1d78ab88e73ea1